api/instance.go

```diff
@@ -56,6 +56,37 @@ func (api *API) waitUntilAllNodesReady(instanceID string) error {
 	}
 }
 
+func (api *API) waitUntilAllNodesReadyWithTimeout(instanceID string, attempt, sleep, timeout int) error {
```

Comment:
Can we name this function `waitWithTimeoutUntilAllNodesConfigured`? 

To avoid confusing with the boolean (on clusters in our backend) named `ready`

Reply:
Yes not a bad idea.